app.js

```
@@ -1,101 +1,94 @@
+        
+                    const parsedFeed = parseFeed(feed)
+        
+                    if (!!parsedFeed.episodes.length) {
+                        await sendFeedToTelegram(parsedFeed, channel)
+                    } else {
+                        log(`${channel} parseFeed`, 'Nothing to upload');
+                    }
+                } catch (error) {
+                    log('Error getting feeds:', error);
+                }
+            }
         } else {
-            logger(false, 'No sources to retrieve')
+            log('No sources to retrieve.')
         }
-    }).catch(err => {
-        logger(false, `Database connection error: ${err.message}`)
-    })
+    } catch (error) {
+        log(`Error getting RSS list from Database`)
+    }
 }
 
 /**
  * Obtener la lista de episodios del feed
  * @param {string} rssUri - The url of the RSS Feed
  * @returns {Promise}
  */
-function getFeed(rssUri) {
-    return new Promise((resolve, reject) => {
-
-        requestP(rssUri)
-        .then(response => {
-
-            if (response) {
-                parseString(response, (err, result) => {
-                    if (err) {
-                        reject(['getFeed' + rssUri, err])
-                    } else {
-                        resolve(result.rss.channel[0])
-                    }
+const getFeed = async rssUri => {
+    return new Promise(async (resolve, reject) => {
+        try {
+            const response = await axios.get(rssUri);
+        
+            if (!!response.data) {
+                parseString(response.data, (err, result) => {
+                    if (err) reject(`getFeed ${rssUri}:\n${err}`);
+                    else resolve(result.rss.channel[0]);
                 })
             } else {
-                reject(['getFeed', 'Unable to fetch feed'])
+                reject(`getFeed Feed ${rssUri} did not return any feeds`);
             }
-        })
+        } catch (error) {
+            reject(error);
```

Here, you can throw the error.